outlines/fsm/regex.py

```diff
@@ -572,13 +571,9 @@ def create_fsm_index_tokenizer(
 
     states_to_token_subsets = create_fsm_index_end_to_end(fsm.fsm_info, vocabulary)
 
-    # Allow transitions to EOS from all terminals FSM states that are
-    # reachable
-    # TODO: Do we really need this anymore?
+    # Allow transitions to EOS from all terminals FSM states that are reachable
```

Comment:
This should be covered by [this function](https://github.com/outlines-dev/outlines/blob/407d76efb1a250bf0ad85f8730c97ccaa06462e2/outlines/fsm/fsm.py#L127). We should understand why it isn't in your case.

Reply:
I can get on board with this, but if I understand well we should be able to remove the `if state in self.final_states: return [eos_token_id]` in `RegexFSM`?